### PR TITLE
For #23159 - Created setting to turn off the Contile feature

### LINKED ISSUE #23159: Add setting for Contile to turn off shortcuts
https://mozilla-hub.atlassian.net/browse/FNXV2-18742

### PR DESCRIPTION
**Pull request by MozillaNoah**
Closes #23159

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1883,8 +1885,9 @@
     <string name="top_sites_max_limit_content_2">To add a new top site, remove one. Touch and hold the site and select remove.</string>
     <!-- Confirmation dialog button text when top sites limit is reached. -->
     <string name="top_sites_max_limit_confirmation_button">OK, Got It</string>
-    <!-- Label for the preference to show the most visited top sites on the homepage -->
-    <string name="top_sites_toggle_top_recent_sites_3">Most visited top sites</string>
+    <!-- Label for the preference to show the shortcuts for the most visited top sites on the homepage -->
+    <string name="top_sites_toggle_top_recent_sites_3" moz:removedIn="98" tools:ignore="UnusedResources">Most visited top sites</string>
+    <string name="top_sites_toggle_top_recent_sites_4">Shortcuts</string>
```
Comment (gabrielluong) [inline app/src/main/res/values/strings.xml:1890]: Needs a comment

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt [hunk 2]
```diff
@@ -1248,4 +1248,13 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         featureFlag = FeatureFlags.isPocketRecommendationsFeatureEnabled(appContext),
         default = { appContext.components.analytics.features.homeScreen.isPocketRecommendationsActive() },
     )
+
+    /**
+     * Indicates if the Contile functionality should be visible.
+     */
+    var showContileFeature by lazyFeatureFlagPreference(
+        appContext.getPreferenceKey(R.string.pref_key_enable_contile),
+        default = { false },
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/utils/Settings.kt:1257]:
```suggestion
        default = false,
```

Looks like this can be `false`

Comment (MozillaNoah) [reply app/src/main/java/org/mozilla/fenix/utils/Settings.kt:1257]: One would think anyway
image

File: app/src/main/res/layout/checkbox_left_sub_preference.xml [hunk 3]
```diff
@@ -12,7 +12,7 @@
     android:clickable="true"
     android:focusable="true"
     android:gravity="center_vertical"
-    android:layout_marginStart="@dimen/custom_checkbox_alignment_margin">
+    android:paddingStart="@dimen/custom_checkbox_alignment_margin">
```
Comment (gabrielluong) [inline app/src/main/res/layout/checkbox_left_sub_preference.xml:15]: I think this makes sense to me, but wondering why did we decide to make this change?

Comment (MozillaNoah) [reply app/src/main/res/layout/checkbox_left_sub_preference.xml:15]: For the clickable area; with margin, this setting and Search -> Show in private sessions were missing considerable tappable area on the left.

Comment (gabrielluong) [reply app/src/main/res/layout/checkbox_left_sub_preference.xml:15]: Normally, we should probably make this change in its own issue/PR.

File: app/src/main/res/values/preference_keys.xml [hunk 4]
```diff
@@ -200,6 +200,8 @@
     <string name="pref_key_top_sites_max_limit" translatable="false">pref_key_top_sites_max_limit</string>
     <!-- Whether or not the top frecent sites are shown along with pinned top sites -->
     <string name="pref_key_enable_top_frecent_sites" translatable="false">pref_key_top_frecent_sites</string>
+    <!-- Whether or not the sponsored shortcuts are shown along with the top recent sites shortcuts -->
+    <string name="pref_key_enable_contile" translatable="false">pref_key_sponsored_shortcuts</string>
```
Comment (gabrielluong) [inline app/src/main/res/values/preference_keys.xml:204]:
```suggestion
    <string name="pref_key_enable_contile" translatable="false">pref_key_enable_contile</string>
```

**Review by gabrielluong — APPROVED** [review]
r+ assuming requested changes are fixed.

**Pull request merged**